python/lsst/ts/vent/controller/dispatcher.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import traceback
+
+from lsst.ts import tcpip
+
+from .controller import Controller
+
+
+def _type_convert(t: type, value: str):
+    """Converts the value string to the specified type. In the case of boolean,
+    "True" or "T" or "1" is ``True`` and everything else is ``False``. Other
+    cases are handled by cast.
+
+    Parameters
+    ----------
+    t: type
+        The type that value should be converted to.
+
+    value: str
+        The value to be converted.
+
+    Raises
+    ------
+    ValueError
+        If the value cannot be converted to the specified type.
+    """
+
+    if t == bool:  # Boolean is a special case
+        return value.lower() in ("true", "t", "1")
+    return t(value)
+
+
+class Dispatcher(tcpip.OneClientReadLoopServer):
+    def __init__(
+        self, port: int, log: logging.Logger, controller: Controller = Controller()
```

Comment:
Are you sure you want to provide a default `Controller` here? I think this will create a weird class signature in Python. If you want to allow users to create a `Dispatcher` without passing a `Controller`, consider doing something like:

```
def __init__(
        self, port: int, log: logging.Logger, controller: Controller | None = None
):

    self.controller = controller if controller is not None else Controller
```

This will ensure a new `Controller` is created for every `Dispatcher` that does not get a `controller`. I think your current implementation might have some weird repercussions.